angr/analyses/find_objects_static.py

```diff
@@ -0,0 +1,130 @@
+import claripy
+from .forward_analysis import FunctionGraphVisitor, SingleNodeGraphVisitor, ForwardAnalysis
+from . import Analysis
+from ..analyses.reaching_definitions.function_handler import FunctionHandler
+from ..knowledge_plugins.key_definitions.atoms import Tmp, Register, MemoryLocation
+from ..storage.memory_mixins.paged_memory.pages.multi_values import MultiValues
+from ..knowledge_plugins.key_definitions.constants import OP_BEFORE, OP_AFTER
+
+
+class PossibleObject():
+    def __init__(self, size, addr, class_name=None):
+        self.size = size
+        # This address is only valid during RDA as we map new objects outside the already mapped region
+        self.addr = addr
+        self.class_name = class_name
+
+    def __hash__(self):
+        return hash((self.addr, self.size, self.class_name))
+
+    def __eq__(self, other):
+        return self.size == other.size and self.addr == other.size and self.class_name == other.class_name
+
+
+class NewFunctionHandler(FunctionHandler):
+    def __init__(self, max_addr=None, new_func_addr=None, cfg=None):
+        self.max_addr = max_addr
+
+        # this is a map between an object addr outside the mapped binary and PossibleObject instance
+        self.possible_objects_dict = {}
+
+        # address of the new() function
+        self.new_func_addr = new_func_addr
+        self.cfg = cfg
+
+    def hook(self, analysis):
+        return self
+
+    def handle_local_function(self, state, function_address, call_stack,
+                              maximum_local_call_depth, visited_blocks, dep_graph,
+                              src_ins_addr=None,
+                              codeloc=None):
+        if function_address == self.new_func_addr:
+            # check if this is a call to new()
+            # reading from rdi for the size argument passed to new()
+            size = state.register_definitions.load(72, state.arch.bits//state.arch.byte_width).values[0].pop()._model_concrete.value
+
+            # None since we do not know it's class yet, it a possible this pointer
+            self.possible_objects_dict[self.max_addr] = PossibleObject(size, self.max_addr)
+
+            # assigning eax a concrete address to track the possible this pointer
+            atom = Register(16, 8)

```

Comment:
1) Can we avoid hardcoding register offsets? libvex makes no guarantee that these offsets are stable from build to build
2) Can we use calling convention objects to avoid hardcoding support for one particular arch?